lib/pbench/server/api/resources/upload_api.py

```diff
@@ -0,0 +1,192 @@
+import os
+import humanize
+import tempfile
+import hashlib
+from pathlib import Path
+from flask_restful import Resource, abort
+from flask import request, jsonify
+from werkzeug.utils import secure_filename
+
+ALLOWED_EXTENSIONS = {"xz"}
+
+
+class HostInfo(Resource):
+    def __init__(self, server_config, logger, api):
+        self.server_config = server_config
+        self.api = api
+        self.logger = logger
+
+    def get(self):
+        try:
+            response = jsonify(
+                dict(
+                    message=f"{self.server_config.get('user')}@{self.server_config.get('host')}"
+                    f":{self.server_config.get('pbench-receive-dir-prefix')}-002"
+                )
+            )
+        except Exception:
+            self.logger.exception(
+                "There was something wrong constructing the host info"
+            )
+            abort(500, message="There was something wrong with your request")
+        response.status_code = 200
+        return response
+
+
+class Upload(Resource):
+    def __init__(self, upload_directory, max_content_length, logger, api):
+        self.upload_directory = upload_directory
+        self.max_content_length = max_content_length
```

Comment:
This item is specific to the `Upload` class, so perhaps we should pass the configuration object so that we don't need to endow knowledge of what the `Upload` class needs outside of the class itself.